app/src/main/kotlin/com/github/gotify/messages/MessagesActivity.kt

```diff
@@ -204,6 +206,14 @@ internal class MessagesActivity :
                 .into(t)
         }
         selectAppInMenu(selectedItem)
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
The MessagesActivity may not have been started when the WebSocketService is running. F.ex. if the service was started via the `com.github.gotify.init.BootCompletedReceiver`.

How about we move the channel creation completely to the WebSocketService? The Service already has an indirect List<Application> via the PicassoHandler see `com.github.gotify.picasso.PicassoHandler#getIcon`. We could move the `Map<int, Application>` from the PicassoHandler to the WebSocketService, then the Service could create the notification channels with the applications.

Reply:
Same reason here: The notification channels are created on the 'onUpdateApps' method.
Is it called on every app start or only when manually refreshing the apps with the refresh button?

Reply:
Is what called on every app start? The Map inside the PicassoHandler is refreshed on com.github.gotify.service.WebSocketService#onStartCommand, the AppHolder from the MessageActivity is refreshed when the activity is started or you manually trigger the refresh in the drawer.

Reply:
That's true. So currently a device reboot does not update the notification channels.  
Opening the app would then do that.

But if we move the channel creation over to the WebSocket service a simple restart of the app would not be enough anymore to update the channels (when turning them on in the settings or when updating the name).  
We would have to reconnect the WebSocket, am I right? (I have no experience with WebSockets on Android)

Reply:
A refresh (`com.github.gotify.messages.MessagesActivity#refreshAll`) inside the navigation drawer starts the InitializationActivity, this activitiy also (re)starts the WebSocketService (`com.github.gotify.init.InitializationActivity#authenticated`). This is visible inside the logs: `Starting WebSocketService` is logged when the refresh is triggered.

TL;DR: `com.github.gotify.service.WebSocketService#onStartCommand` will be called when the app is started via the BootCompleteReceiver, when normally starting the app, or when a refreshAll is triggered via the drawer.

Reply:
Nice, didn't know that.

Reply:
Wait, I'm blind.  
Where does the PicassoHandler hold the mentioned `Map<int, Application>`? I can only find the `appIdToAppImage` map of type `<Long, String>` holding the app ids and icon urls.

Reply:
Yeah it's only Map<Long, String> but it has access to all applications in here com.github.gotify.picasso.PicassoHandler#updateAppIds, so it can be easily changed to something like Map<Long, Application>.

Reply:
Something like this? 1182f358cba3f450b5ed3413840e2dc46af39bd4

Reply:
Yes, but the map could be moved to the WebSocketService, because it's not really needed inside the PicassoHandler, I've fixed this already in my commits.

Reply:
Then I've misunderstood something.
Thank you